app/src/main/java/com/vestrel00/daggerbutterknifemvp/ui/main/MainActivityModule.java

```diff
@@ -43,6 +47,26 @@
     @ContributesAndroidInjector(modules = MainFragmentModule.class)
     abstract MainFragment mainFragmentInjector();
 
+    /**
+     * Provides the injector for the {@link Example1Fragment}, which has access to the dependencies
+     * provided by this activity and application instance (singleton scoped objects).
+     *
+     * This is used for example 4, which displays {@link Example1Fragment} as a {@link android.app.DialogFragment}.
+     */
+    @PerFragment
+    @ContributesAndroidInjector(modules = Example1FragmentModule.class)
+    abstract Example1Fragment example1FragmentInjector();
```

Comment:
The `MainActivity` now creates `Example1Fragment` directly (previously it was only created by `Example1Activity`). Therefore, the `MainActivityModule` must also provide the injector for `Example1Fragment` (as `Example1Activity` does).